openshift/minishift-addon/rhche/rhche.addon

```diff
@@ -0,0 +1,38 @@
+# Name: rhche
+# Description: Setup and Configure Red Hat Che Template
+# Url: https://www.eclipse.org/che/docs/setup/openshift/index.html
+# NAMESPACE, CHE_DOCKER_IMAGE, OPENSHIFT_TOKEN, GITHUB_CLIENT_ID, GITHUB_CLIENT_SECRET
+# NAMESPACE=mini-che, CHE_DOCKER_IMAGE=eclipse/che-server:latest, OPENSHIFT_TOKEN=NULL, GITHUB_CLIENT_ID=changeme, GITHUB_CLIENT_SECRET=changeme
+# Required-Vars: NAMESPACE, CHE_DOCKER_IMAGE, CHE_VERSION, KEYCLOAK_DOCKER_IMAGE
+# Var-Defaults: NAMESPACE=rhche, CHE_DOCKER_IMAGE=registry.devshift.net/che/rh-che-server, CHE_VERSION=latest, KEYCLOAK_DOCKER_IMAGE=eclipse/che-keycloak
+# OpenShift-Version: >=3.5.0
+
+echo [CHE] Create the Che server Template
+oc apply -f templates/rh-che.app.yaml -n openshift
+
+echo [CHE] Creating #{NAMESPACE} project as developer
+oc adm new-project #{NAMESPACE} --description="Red Hat Che on minishift"
+oc adm policy add-role-to-user admin developer -n #{NAMESPACE}
+
+echo [CHE] Deploying Postgres
+oc new-app -f templates/multi/postgres-template.yaml -n #{NAMESPACE}
+
+echo [CHE] Waiting 30s for Keycloak to be up and running
+sleep 30
+
+echo [CHE] Deploying Keycloak
+oc new-app -f templates/multi/keycloak-template.yaml -p IMAGE_KEYCLOAK=#{KEYCLOAK_DOCKER_IMAGE} -p ROUTING_SUFFIX=#{routing-suffix} -n #{NAMESPACE}
+
+echo [CHE] Waiting 60s for Keycloak to be up and running
+sleep 60
```

Comment:
I have taken in consideration your feedback and I have splitted the addon in 2 separate addons (I have not put all in one yaml because you may want to use an existing keycloak/postgre).